### PR TITLE
[rebase-dts] Import DTS for MSM8953/MSM8956/MSM8976 devices

### PR DESCRIPTION
**Pull request by Kiciuk**
This pull request bring device trees from msm8953 lk2nd fork

### CODE REVIEW

File: lk2nd/device/dts/msm8953/msm8953-meizu-m1721.dts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_MSM8953 0>;
+	qcom,board-id = <0xb 0>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-meizu-m1721.dts:10]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID_QRD 0>;
```

and same as kenzo

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-daisy.dts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_MSM8953 0>;
+	qcom,board-id = <0x1000b 0x09>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-xiaomi-daisy.dts:10]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID(MTP, 1, 0) 0x09>;
```

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-markw.dts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_MSM8953 0>;
+	qcom,board-id = <0x1000b 1>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-xiaomi-markw.dts:10]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID(MTP, 1, 0) 1>;
```

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-vince.dts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_MSM8953 0>;
+	qcom,board-id = <0x1000b 8>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-xiaomi-vince.dts:10]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID(MTP, 1, 0) 8>;
```

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-ysl.dts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_MSM8953 0>;
+	qcom,board-id = <8 0>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-xiaomi-ysl.dts:10]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID_MTP 0>;
```
and same as kenzo

File: lk2nd/device/dts/msm8953/sdm450-samsung-r04.dts [hunk 6]
```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_SDM450 0>;
+	qcom,board-id = <8 4>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/sdm450-samsung-r04.dts:10]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID_MTP 4>;
```

File: lk2nd/device/dts/msm8953/sdm450-samsung-r05.dts [hunk 7]
```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_SDA450 0>;
+	qcom,board-id = <8 5>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/sdm450-samsung-r05.dts:10]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID_MTP 5>;
```

File: lk2nd/device/dts/msm8953/sdm632-fairphone-fp3.dts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	model = "Fairphone 3";
+	qcom,msm-id = <QCOM_ID_SDM632 0>;
+	qcom,pmic-id = <0x10016 0x25 0x00 0x00>;
+	qcom,board-id = <8 3>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/sdm632-fairphone-fp3.dts:12]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID_MTP 3>;
```

File: lk2nd/device/dts/msm8953/sdm632-fairphone-fp3.dts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+	__symbols__ {};
+	chosen {};
+	memory {};
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/sdm632-fairphone-fp3.dts:15]:
```suggestion
```

they are included in skeleton64.dtsi:
https://github.com/msm8916-mainline/lk2nd/blob/6461706bb92978923474ac670ef38249dbba4e95/lk2nd/device/dts/skeleton64.dtsi#L9-L14

File: lk2nd/device/dts/msm8953/sdm632-motorola-ocean.dts [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+#include <motorola-carrier-channel-ids.dtsi>
+
+/ {
+	model = "Qualcomm Technologies, Inc. SDM632 + PMI632 MTP S3";
+	qcom,msm-name = "SDM632";
+	qcom,msm-id = <QCOM_ID_SDM632 0>;
+	qcom,pmic-id = <0x10016 0x25 0x00 0x00>;
+	qcom,board-id = <8 3>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/sdm632-motorola-ocean.dts:14]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID_MTP8 3>;
```

How does this not clash with fp3? Do they use model= or something?

File: lk2nd/device/dts/msm8953/sdm632-motorola-ocean.dts [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+	__symbols__ {};
+	chosen {};
+	memory {};
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/sdm632-motorola-ocean.dts:17]:
```suggestion
```

File: lk2nd/device/dts/msm8953/sdm632-xiaomi-onclite.dts [hunk 12]
```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_SDM632 0>;
+	qcom,board-id = <8 0>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/sdm632-xiaomi-onclite.dts:10]:
```suggestion
	qcom,board-id = <QCOM_BOARD_ID_MTP 0>;
```

And this is also same question as kenzo

File: lk2nd/device/dts/msm8953/msm8953-huawei-milan.dts [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+			<8017 0>, <8018 0>, <8019 0>, <8032 0>, <8033 0>,
+			<8035 0>, <8048 0>, <8049 0>, <8050 0>, <8051 0>,
+			<8066 0>, <8064 0>, <8067 0>, <8080 0>, <8082 0>,
+			<8083 0>, <8098 0>, <8099 0>, <8114 0>, <8130 0>;
+};
+
+&lk2nd {
+	model = "Huawei Maimang 5 / Nova (Plus) / G9 (Plus)";
+	compatible = "huawei,milan", "qcom,msm8953", "lk2nd,device";
+
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-huawei-milan.dts:21]: Could you please annotate all of them with (close-to-)mainline dts file names from 8953 fork? This way lk2nd will be able to pick it automatically in generic images later.

```suggestion

	lk2nd,dtb-files = "msm8953-huawei-milan";
	
```

If there is no obvious dts, I usually added a fixme and guessed the name like this:

https://github.com/msm8916-mainline/lk2nd/blob/6461706bb92978923474ac670ef38249dbba4e95/lk2nd/device/dts/msm8916/msm8916-asus-z00e.dts#L15

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-daisy.dts [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+/ {
+	qcom,msm-id = <QCOM_ID_MSM8953 0>;
+	qcom,board-id = <QCOM_BOARD_ID(MTP, 1, 0) 0x09>;
+};
+
+&lk2nd {
+	model = "Xiaomi Mi A2 Lite";
+	compatible = "xiaomi,daisy";
+	lk2nd,match-bootloader = "*TISSOT*";
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-xiaomi-daisy.dts:16]: hm is "daisy" presenting itself as "tissot"? the same way as the "tissot" in mi a1?

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-common.dts [hunk 15]
```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <QCOM_ID_MSM8953 0>;
+	qcom,board-id = <0x1000b 0>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-xiaomi-common.dts:10]: this is (QRD, 1, 0) afaict

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-ysl.dts [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+/ {
+	qcom,msm-id = <QCOM_ID_MSM8953 0>;
+	qcom,board-id = <QCOM_BOARD_ID_MTP 0>;
+};
+
+&lk2nd {
+	model = "Xiaomi Redmi S2/Y2 (ysl)";
+	compatible = "xiaomi,ysl";
+
+	panel {
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/msm8953-xiaomi-ysl.dts:17]: this looks like a leftover duplicate of msm8953-mtp.dts now

File: lk2nd/device/dts/msm8953/sdm632-fairphone-fp3.dts [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+	model = "Fairphone 3";
+	qcom,msm-id = <QCOM_ID_SDM632 0>;
+	qcom,pmic-id = <0x10016 0x25 0x00 0x00>;
+	qcom,board-id = <QCOM_BOARD_ID_MTP 3>;
+	__symbols__ {};
+};
+
+&lk2nd {
+	model = "Fairphone 3";
+	compatible = "fairphone,fp3", "qcom,msm8953", "lk2nd,device";
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8953/sdm632-fairphone-fp3.dts:18]:
```suggestion
	compatible = "fairphone,fp3";	
```

**Review by TravMurav — COMMENTED** [review]
One other thing I'd appreciate if you do: Could you add a list of contributors to the original files in the commit descriptions? I was generating them like this:

```sh
git shortlog -nse origin/master -- dts/msm8974/msm8974-sa* | sed 's/^.*\t/Co-authored-by: /'
```

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-daisy.dts (hunk 14 above)
Comment (Kiciuk) [reply lk2nd/device/dts/msm8953/msm8953-xiaomi-daisy.dts:16]: `androidboot.bootloader=MSM8953_DAISY2.0_20190717191738`
From console-ramoops of mi a2 lite so it seems so

File: lk2nd/device/dts/msm8953/msm8953-xiaomi-ysl.dts (hunk 16 above)
Comment (Kiciuk) [reply lk2nd/device/dts/msm8953/msm8953-xiaomi-ysl.dts:17]: Dropped from rules yet still on filesystem indeed.

**Comment by TravMurav** [comment]
Great job, thanks!

**Pull request merged**